onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -850,15 +852,30 @@ def _infer_Reshape(self, node):
 
             assert new_sympy_shape.count(-1) < 2
             if -1 in new_sympy_shape:
-                new_dim = total // non_deferred_size
+                # handle a special case where input_shape is [] and shape_value is [-1]
+                if len(input_shape) == 0 and len(shape_value) == 1:
+                    if node.output[0] in self.sympy_data_:
```

Comment:
>node.output[0] in self.sympy_data_ [](start = 23, length = 34)

will this happen?

Reply:
I am actually not sure if this will happen or not. I just put it there in case it would happen. We could replace the ```if``` with ```assert``` to ensure our assumption.